GliaWidgets/Sources/QueuesMonitor/QueuesMonitor.swift

```diff
@@ -0,0 +1,78 @@
+import Foundation
+import GliaCoreSDK
+import Combine
+
+class QueuesMonitor {
+    enum State {
+        case idle
+        case updated([Queue])
+        case failed(Error)
+    }
+
+    @Published private(set) var state: State = .idle
+
+    private let environment: Environment
+    private var subscriptionId: String?
+    private var queues: [Queue] = []
+
+    init(environment: Environment) {
+        self.environment = environment
+
+        startMonitoring()
+    }
+
+    func startMonitoring() {
+        environment.sdkClient.listQueues { [weak self] queues, error in
+            guard let self else {
+                return
+            }
+            if let error {
+                self.state = .failed(error)
+                return
+            }
+
+            if let queues {
+                self.queues = queues
+                self.state = .updated(queues)
+                self.observeQueuesUpdates(queues)
+                return
+            }
+        }
+    }
+
+    func stopMonitoring() {
+        if let subscriptionId {
+            environment.sdkClient.unsubscribeFromUpdates(subscriptionId) { [weak self] error in
+                self?.state = .failed(error)
+            }
+        }
+    }
+}
+
+private extension QueuesMonitor {
+    func updateQueue(_ queue: Queue) {
```

Comment:
Just to be sure that this logic is correct, I think it makes sense to add tests here as well.

Reply:
Since unit tests will be added in further PRs, I'd add some TODO